sos/report/plugins/gluster.py

```diff
@@ -41,15 +41,16 @@ def wait_for_statedump(self, name_dir):
                     ret = string.count(last_line, 'DUMP_END_TIME')
 
     def postproc(self):
-        if not os.path.exists(self.statedump_dir):
-            return
-        try:
-            for dirs in os.listdir(self.statedump_dir):
-                os.remove(os.path.join(self.statedump_dir, dirs))
-            os.rmdir(self.statedump_dir)
-            os.unlink('/tmp/glusterdump.options')
-        except OSError:
-            pass
+        if self.get_option("dump"):
+            if not os.path.exists(self.statedump_dir):
+                return
+            try:
+                for name in os.listdir(self.statedump_dir):
+                    if 'dump' in name:
```

Comment:
Please use the full pattern we discussed on IRC: `*.dump.*`: we know that all Gluster dump files conform to this pattern, so it is the most strict we can make it without adding cumbersome steps like PID parsing and checking the timestamp portion falls in a recent window.

Reply:
I suggest we use .*dump.[0-9]*, statedump files end with dump.epoch time

Reply:
it should be .\*dump.[0-9]\*

Reply:
.*dump.[0-9]* not working due to glob syntax used 

                for name in glob.glob(self.statedump_dir + '/*.dump.[0-9]*'):
                    os.remove(name)